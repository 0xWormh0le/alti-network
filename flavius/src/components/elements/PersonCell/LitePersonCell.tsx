import Tooltip from 'components/widgets/Tooltip'
import React from 'react'
import { Link } from 'react-router-dom'
import { modalBasePath, truncateText } from 'util/helpers'
import './PersonCell.scss'

export interface LitePersonCellProps {
  person: LitePerson
  platformId: string
}

const LitePersonCell: React.FC<LitePersonCellProps> = ({ person, platformId }) => {
  return (
    <div className='PersonCell'>
      <Link
        className='PersonCell__email'
        to={`${modalBasePath()}/spotlight/${encodeURIComponent(
          person.primaryEmail?.address ?? ''
        )}?platformId=${platformId}`}>
        <div className='PersonCell__email'>
          <Tooltip text={person.primaryEmail?.address ?? ''} id='tooltip-primary-email-id'>
            <span>{truncateText(person.primaryEmail?.address ?? '', 20)}</span>
          </Tooltip>
        </div>
      </Link>
    </div>
  )
}

export default LitePersonCell
