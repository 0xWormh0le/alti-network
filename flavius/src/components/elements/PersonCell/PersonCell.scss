@import 'styles/_shared.scss';

.PersonCell {
  &__link {
    @include brand-link();

    &:hover {
      .PersonCell__email {
        color: $AZURE;
      }
    }
  }

  &__full-name {
    display: flex;
  }

  &__name {
    max-width: 100px;
    height: 14px;
    line-height: normal;
    white-space: nowrap;
    overflow: hidden;
    text-overflow: ellipsis;
  }

  &__email {
    @include set-typography('BODY', 'normall');
    color: $TEXT_COLOR;
  }

  &__email-ext {
    margin-left: 0.5em;
  }
}
