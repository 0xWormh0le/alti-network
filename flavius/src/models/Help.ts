import UI_STRINGS from 'util/ui-strings'

export interface IHelpQuestion {
  question: string
  risk_ids: RiskTypeId[]
}

export const HelpQuestionsList: IHelpQuestion[] = [
  {
    question: UI_STRINGS.HELP.SHARING_RISKS,
    risk_ids: [1011, 1013, 1020, 1021, 1022, 1023]
  },
  {
    question: UI_STRINGS.HELP.RELATIONSHIP_RISKS,
    risk_ids: [2000, 2001, 2002, 2003]
  },
  {
    question: UI_STRINGS.HELP.ACTIVITY_RISKS,
    risk_ids: [3100, 3200, 3010, 3012]
  },
  {
    question: UI_STRINGS.HELP.INFORMATIONAL_RISKS,
    risk_ids: [0, 10, 11]
  }
]
