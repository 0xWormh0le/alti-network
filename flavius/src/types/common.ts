export enum UsageKind {
  personal = 'personal',
  work = 'work'
}

export enum UserKind {
  user = 'user',
  person = 'person',
  serviceAccount = 'serviceAccount'
}

export enum AccessLevel {
  member = 'member',
  manager = 'manager',
  owner = 'owner',
  admin = 'admin'
}

export enum RiskCategoryType {
  SHARING_RISKS = 'Sharings',
  RELATIONSHIP_RISKS = 'Relationships',
  ACTIVITY_RISKS = 'Actions',
  INFORMATIONAL_RISKS = 'Informatives'
}

export enum FileActivityType {
  ACCESS_GRANTED = 'ACCESS_GRANTED',
  CHANGE_FOLDER_PERMISSION = 'CHANGE_FOLDER_PERMISSION',
  CONTENT_ACCESS = 'CONTENT_ACCESS',
  DOWNLOAD = 'DOWNLOAD',
  FILE_WATERMARKED_DOWNLOAD = 'FILE_WATERMARKED_DOWNLOAD',
  ITEM_DOWNLOAD = 'ITEM_DOWNLOAD',
  ITEM_PREVIEW = 'ITEM_PREVIEW',
  ITEM_SHARED = 'ITEM_SHARED',
  ITEM_SHARED_CREATE = 'ITEM_SHARED_CREATE',
  ITEM_SHARED_UPDATE = 'ITEM_SHARED_UPDATE',
  ITEM_SYNC = 'ITEM_SYNC',
  SHARE = 'SHARE'
}

export enum PermissionRole {
  own = 'own',
  read = 'read',
  write = 'write'
}

export enum PermissionShared {
  external = 'external',
  internal = 'internal',
  none = 'none'
}

export enum PermissionType {
  anyone = 'anyone',
  domain = 'domain',
  user = 'user'
}

export enum PermissionItemStatus {
  active = 'ACTIVE',
  pending = 'PENDING',
  removed = 'REMOVED',
  cannotBeRemoved = 'CANNOT_BE_REMOVED'
}

export enum SensitiveContentStatus {
  DETECTION_PENDING = 'pending',
  FOUND_SENSITIVE_CONTENT = 'yes',
  FOUND_NO_SENSITIVE_CONTENT = 'no',
  FAILED_DETECTION = 'failed'
}

export enum FileGridNavType {
  TIMELINE = 'timeline',
  INTERNAL = 'internal',
  EXTERNAL = 'external',
  SENSITIVE = 'sensitive',
  FILES_IN_FOLDER = 'filesInFolder',
  DETAILS = 'details'
}
