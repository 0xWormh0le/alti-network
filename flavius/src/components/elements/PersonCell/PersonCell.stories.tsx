import React from 'react'
import { Story, Meta } from '@storybook/react/types-6-0'
import PersonCell, { PersonCellProps } from './PersonCell'
import { BrowserRouter } from 'react-router-dom'
import { UsageKind, UserKind, AccessLevel } from 'types/common'

export default {
  title: 'Elements/PersonCell',
  component: PersonCell
} as Meta

const MULTIPLE_OWNER_FILE_MOCK: FileOwner = {
  name: {
    givenName: 'Joe',
    familyName: 'Schmough',
    fullName: 'Joe Schmough'
  },
  primaryEmail: {
    address: 'joe@thoughtlabs.io',
    kind: UsageKind.personal,
    primary: true,
    accessCount: 0,
    riskCount: 0,
    lastDeletedPermissions: 0
  },
  recoveryEmail: {
    address: 'test@email.com',
    kind: UsageKind.personal,
    primary: true,
    accessCount: 0,
    riskCount: 0,
    lastDeletedPermissions: 0
  },
  avatar: {
    url: 'http://i.pravatar.cc/200?img=15',
    url_etag: 'http://i.pravatar.cc/200?img=15'
  },
  internalCount: 1,
  externalCount: 0,
  emails: [],
  internal: true,
  accessCount: 0,
  riskCount: 99,
  riskTypeId: 0,
  fileCount: 2,
  altnetId: '',
  projectId: '',
  accessLevel: AccessLevel.member,
  userKind: UserKind.person,
  phones: [],
  externalIds: [],
  orgUnitPath: '',
  etag: '',
  isEnrolledInMFA: false,
  creationTime: 0,
  lastLoginTime: 0,
  lastModifiedTime: 0,
  notes: {
    content: '',
    content_type: ''
  }
}

const MULTIPLE_OWNER_THOUSANDS_RISK_TYPE_FILE_MOCK: FileOwner = {
  name: {
    givenName: 'Joe-long-name-example-for-test',
    familyName: 'Schmough',
    fullName: 'Joe Schmough'
  },
  primaryEmail: {
    address: 'joe@thoughtlabs.io',
    kind: UsageKind.personal,
    primary: true,
    accessCount: 0,
    riskCount: 0,
    lastDeletedPermissions: 0
  },
  recoveryEmail: {
    address: 'test@email.com',
    kind: UsageKind.personal,
    primary: true,
    accessCount: 0,
    riskCount: 0,
    lastDeletedPermissions: 0
  },
  avatar: {
    url: 'http://i.pravatar.cc/200?img=15',
    url_etag: 'http://i.pravatar.cc/200?img=15'
  },
  internalCount: 1,
  externalCount: 0,
  emails: [],
  internal: true,
  accessCount: 0,
  riskCount: 99,
  riskTypeId: 1020,
  fileCount: 1,
  altnetId: '',
  projectId: '',
  accessLevel: AccessLevel.member,
  userKind: UserKind.user,
  phones: [],
  externalIds: [],
  orgUnitPath: '',
  etag: '',
  isEnrolledInMFA: false,
  creationTime: 0,
  lastLoginTime: 0,
  lastModifiedTime: 0,
  notes: {
    content: '',
    content_type: ''
  }
}

const Template: Story<PersonCellProps> = (args) => (
  <BrowserRouter>
    <PersonCell {...args} />
  </BrowserRouter>
)

export const MultipleOwnerThousandsRiskType = Template.bind({})
MultipleOwnerThousandsRiskType.args = {
  personData: MULTIPLE_OWNER_THOUSANDS_RISK_TYPE_FILE_MOCK
}

export const MultipleFileOwner = Template.bind({})
MultipleFileOwner.args = {
  personData: MULTIPLE_OWNER_FILE_MOCK
}
