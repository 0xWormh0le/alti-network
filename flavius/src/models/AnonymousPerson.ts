import Person from './Person'
import { UsageKind } from 'types/common'

const email: Email = {
  address: 'sample@foo.bar',
  kind: UsageKind.personal,
  primary: false,
  accessCount: 1,
  riskCount: 1,
  lastDeletedPermissions: 0
}

class AnonymousPerson extends Person {
  constructor() {
    super({
      name: {
        familyName: 'User',
        givenName: 'Anonymous',
        fullName: 'User Anonymous'
      },
      primaryEmail: email
    } as IPerson)
  }
}

export default AnonymousPerson
