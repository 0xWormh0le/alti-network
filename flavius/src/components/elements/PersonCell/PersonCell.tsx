import React, { useContext } from 'react'
import { Link } from 'react-router-dom'
import Person from 'models/Person'
import { isExternalEmail, modalBasePath, searchWithoutModalParams, truncateText } from 'util/helpers'
import { SharingRiskTypeIds } from 'models/RiskCatalog'
import { Tooltip } from 'components/widgets/Tooltip/Tooltip'
import Marker, { MarkerType } from 'components/elements/Marker'
import { UI_STRINGS } from 'util/ui-strings'
import { UserContext } from 'models/UserContext'
import Typography, { TypographyVariant } from '../Typography'
import './PersonCell.scss'

export interface PersonCellProps {
  personData: IPerson & { fileCount?: number; riskTypeId?: RiskTypeId }
}

export const PersonCell: React.FC<PersonCellProps> = ({ personData }) => {
  const user = useContext(UserContext)
  const person = new Person(personData)
  if (!person) return null
  const fileCount = personData.fileCount || 0
  const riskTypeId = personData.riskTypeId || 0
  const isExternal = isExternalEmail(user.domains, person?.primaryEmail ? person.primaryEmail.address ?? '' : '')
  // const displayUsername: boolean = Boolean(person && person?.displayName !== UI_STRINGS.RISKS.ANONYMOUS_USER)
  const { BODY } = TypographyVariant

  return (
    <div className='PersonCell'>
      {fileCount > 1 && SharingRiskTypeIds.indexOf(riskTypeId) === -1 ? (
        <div className='PersonCell__full-name'>
          <Tooltip text={UI_STRINGS.RISKS.TOOLTIP_MULTIPLE} id={`tooltip-multiple-id`}>
            <span>{UI_STRINGS.RISKS.MULTIPLE}</span>
          </Tooltip>
        </div>
      ) : (
        person &&
        (person.displayName && !person.anonymous ? (
          <Link
            to={`${modalBasePath()}/spotlight/${encodeURIComponent(
              person.primaryEmail?.address ?? ''
            )}${searchWithoutModalParams()}`}
            className='PersonCell__link'>
            {/* {displayUsername && (
              <div className='PersonCell__full-name'>
                <Typography variant={BODY} component='span' className='PersonCell__name'>
                  {person.displayName}
                </Typography>
                {isExternal && externalLabel}
              </div>
            )} */}
            <div className='PersonCell__email'>
              <Tooltip text={person.primaryEmail?.address ?? ''} id='tooltip-primary-email-id'>
                <span>{truncateText(person.primaryEmail?.address ?? '', isExternal ? 17 : 20)}</span>
              </Tooltip>
              {isExternal && (
                <Tooltip text={UI_STRINGS.SPOTLIGHT.EXTERNAL_TOOLTIP} id={`tooltip-ex-label-id`}>
                  <Marker type={MarkerType.Ext} className='PersonCell__email-ext' />
                </Tooltip>
              )}
            </div>
          </Link>
        ) : (
          <div className='PersonCell__full-name'>
            <Tooltip text={UI_STRINGS.RISKS.TOOLTIP_SHARED_DRIVE} id='tooltip-shared-id'>
              <Typography variant={BODY} component='span' className='PersonCell__name'>
                {UI_STRINGS.SPOTLIGHT.SHARED_DRIVE}
              </Typography>
            </Tooltip>
          </div>
        ))
      )}
    </div>
  )
}

export default PersonCell
