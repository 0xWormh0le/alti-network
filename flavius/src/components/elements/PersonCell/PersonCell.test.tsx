import React from 'react'
import { renderWithRouter } from 'test/support/helpers'
import { PersonCell, PersonCellProps } from './PersonCell'

import { UsageKind } from 'types/common'
import { toJs } from 'util/helpers'
import { UserContextProvider } from 'models/UserContext'

const avatar = {
  url: 'http://i.pravatar.cc/200?img=15',
  url_etag: 'http://i.pravatar.cc/200?img=15'
}

const mail = {
  address: 'joe@thoughtlabs.io',
  kind: UsageKind.personal,
  primary: true,
  accessCount: 0,
  riskCount: 0,
  lastDeletedPermissions: 0
}

const MULTIPLE_OWNER_FILE_MOCK: FileOwner & IPerson = {
  name: {
    givenName: 'Joe',
    familyName: 'Schmough',
    fullName: 'Joe Schmough'
  },
  primaryEmail: mail,
  recoveryEmail: mail,
  avatar,
  internalCount: 1,
  externalCount: 0,
  emails: [],
  internal: true,
  accessCount: 0,
  riskCount: 99,
  riskTypeId: 20,
  fileCount: 2,
  altnetId: '',
  projectId: '',
  accessLevel: 0,
  userKind: 1,
  phones: [],
  externalIds: [],
  orgUnitPath: '',
  etag: '',
  isEnrolledInMFA: false,
  creationTime: 0,
  lastLoginTime: 0,
  lastModifiedTime: 0,
  notes: {
    content: '',
    content_type: ''
  }
}

const MULTIPLE_OWNER_THOUSANDS_RISK_TYPE_FILE_MOCK: FileOwner = {
  name: {
    givenName: 'Joe-long-name-example-for-test',
    familyName: 'Schmough',
    fullName: 'Joe Schmough'
  },
  primaryEmail: mail,
  recoveryEmail: mail,
  avatar,
  internalCount: 1,
  externalCount: 0,
  emails: [],
  internal: true,
  accessCount: 0,
  riskCount: 99,
  riskTypeId: 1020,
  fileCount: 0,
  altnetId: '',
  projectId: '',
  accessLevel: 0,
  userKind: 1,
  phones: [],
  externalIds: [],
  orgUnitPath: '',
  etag: '',
  isEnrolledInMFA: false,
  creationTime: 0,
  lastLoginTime: 0,
  lastModifiedTime: 0,
  notes: {
    content: '',
    contentType: ''
  }
}

const ANONYMOUS_PERSON: FileOwner = {
  ...MULTIPLE_OWNER_FILE_MOCK,
  name: null,
  primaryEmail: null,
  riskTypeId: null,
  fileCount: 0
}

describe('PersonCell', () => {
  it('renders correctly', () => {
    const props: PersonCellProps = {
      personData: MULTIPLE_OWNER_THOUSANDS_RISK_TYPE_FILE_MOCK
    }
    const { container } = renderWithRouter(<PersonCell {...props} />)
    expect(container).toMatchSnapshot()
  })
  it('renders the string "Mutiple" for multiple file owners', () => {
    const props: PersonCellProps = {
      personData: toJs(MULTIPLE_OWNER_FILE_MOCK)
    }
    const { container } = renderWithRouter(<PersonCell {...props} />)
    expect(container.querySelector('.PersonCell__full-name')).toHaveTextContent('Multiple')
  })

  it('renders the owner for mult-file risks grouped by file owner', () => {
    const props: PersonCellProps = {
      personData: MULTIPLE_OWNER_THOUSANDS_RISK_TYPE_FILE_MOCK
    }
    const { container } = renderWithRouter(<PersonCell {...props} />)
    expect(container.querySelector('.PersonCell__email')?.clientWidth).toBeLessThan(100)
    // expect(container.querySelector('.PersonCell__name')?.clientWidth).toBeLessThan(100)
    // expect(container.querySelector('.PersonCell__full-name')).toHaveTextContent(
    //   'Joe-long-name-example-for-test Schmough'
    // )
  })

  it('renders shared drive if the person is anonymous', () => {
    const props: PersonCellProps = {
      personData: ANONYMOUS_PERSON
    }
    const { container } = renderWithRouter(<PersonCell {...props} />)
    expect(container.querySelector('.PersonCell__full-name')).toHaveTextContent('Shared Drive')
  })

  // Is external by default since domains will be []
  it('renders external marker if external', () => {
    const props: PersonCellProps = {
      personData: MULTIPLE_OWNER_THOUSANDS_RISK_TYPE_FILE_MOCK
    }
    const { container } = renderWithRouter(<PersonCell {...props} />)
    expect(container.querySelectorAll('.PersonCell__email-ext').length > 0).toBeTruthy()
  })

  it('does not render external marker if not external', () => {
    const props: PersonCellProps = {
      personData: MULTIPLE_OWNER_THOUSANDS_RISK_TYPE_FILE_MOCK
    }
    const { container } = renderWithRouter(
      <UserContextProvider value={{ domains: ['thoughtlabs.io'] } as any}>
        <PersonCell {...props} />
      </UserContextProvider>
    )
    expect(container.querySelectorAll('.PersonCell__email-ext').length > 0).toBeFalsy()
  })
})
