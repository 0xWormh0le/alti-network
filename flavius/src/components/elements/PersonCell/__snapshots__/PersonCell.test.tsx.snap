// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PersonCell renders correctly 1`] = `
<div>
  <div
    class="PersonCell"
  >
    <a
      class="PersonCell__link"
      href="/dashboard/spotlight/joe%40thoughtlabs.io"
    >
      <div
        class="PersonCell__email"
      >
        <span
          class="trigger"
        >
          joe@thoughtlabs.i...
        </span>
        <span
          class="Typography Typography--body PersonCell__email-ext trigger Marker Marker__ext"
        >
          EXT
        </span>
      </div>
    </a>
  </div>
</div>
`;
